processor/src/chiplets/aux_trace/mod.rs

```diff
@@ -242,23 +170,111 @@ where
     }
 }
 
-/// Constructs the inclusions to the kernel procedure table.
-fn chiplets_kernel_table_include<E>(main_trace: &MainTrace, alphas: &[E], row: usize) -> E
+/// Builds the inclusions to the kernel procedure table at `row`.
+fn build_kernel_procedure_table_inclusions<E>(main_trace: &MainTrace, alphas: &[E], row: usize) -> E
 where
     E: FieldElement<BaseField = Felt>,
 {
-    if main_trace.is_kernel_row(row) && main_trace.is_addr_change(row) {
-        alphas[0]
-            + alphas[1].mul_base(main_trace.addr(row))
-            + alphas[2].mul_base(main_trace.chiplet_kernel_root_0(row))
-            + alphas[3].mul_base(main_trace.chiplet_kernel_root_1(row))
-            + alphas[4].mul_base(main_trace.chiplet_kernel_root_2(row))
-            + alphas[5].mul_base(main_trace.chiplet_kernel_root_3(row))
+    let selector0 = main_trace.chiplet_selector_0(row);
+    let selector1 = main_trace.chiplet_selector_1(row);
+    let selector2 = main_trace.chiplet_selector_2(row);
+
+    if selector0 == ONE && selector1 == ONE && selector2 == ONE {
+        let addr = main_trace.chiplet_kernel_addr(row);
+        let addr_nxt = main_trace.chiplet_kernel_addr(row + 1);
+        let addr_delta = addr_nxt - addr;
+        let root0 = main_trace.chiplet_kernel_root_0(row);
+        let root1 = main_trace.chiplet_kernel_root_1(row);
+        let root2 = main_trace.chiplet_kernel_root_2(row);
+        let root3 = main_trace.chiplet_kernel_root_3(row);
+
+        let v = alphas[0]
+            + alphas[1].mul_base(addr)
+            + alphas[2].mul_base(root0)
+            + alphas[3].mul_base(root1)
+            + alphas[4].mul_base(root2)
+            + alphas[5].mul_base(root3);
```

Comment:
The [docs](https://0xpolygonmiden.github.io/miden-vm/design/chiplets/kernel_rom.html#kernel-rom-trace) mention that the table has 6 columns, but this has 5 (i.e. `selector0` is not in here but is in the table in the docs). Should we update the code, or the docs?